.github/workflows/update-major-tag.yml

```diff
@@ -0,0 +1,39 @@
+name: Update major version tag
+run-name: Set ${{ github.event.inputs.major_version }} to point to ${{ github.event.inputs.target }}
+
+# Each time there is a minor release, this workflow can be manually run in order to update the major release tag to point to the latest release.
+# Can also be used for rollback if the latest release is found to have problems.
+
+on:
+  workflow_dispatch:
+    inputs:
+      target:
+        description: The tag or reference to use
+        required: true
+      major_version:
+        type: choice
+        description: The major version to update
+        options:
+          - v1-alpha
+          - v1
+        # add items to this list when new major releases are released
+
+jobs:
+  tag:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # required to commit to the repo
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Git config
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
```

Comment:
Do you have a link to a doc which explains how this works?

Reply:
I wish there was something in the docs about this! I failed to find anything relevant. The best I can suggest is [the conversation on this PR](https://github.com/actions/checkout/pull/1184) for actions/checkout.

I suspect we can set anything as the git user and the tagging still works, I believe the choice is just to get the correct image when commits are view on gh webpages.